### PR TITLE
Listen to "charge.failed" Stripe events and send emails

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-server/issues/15837
~~:warning: depends on https://github.com/mozilla/addons-server/pull/15918~~

### CODE REVIEW

**Comment by willdurand** [comment]
rebased!

**Comment by willdurand** [comment]
(rebased)

File: src/olympia/promoted/tasks.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            event_type,
+            event_id,
+            charge_id,
+        )
+        return
+
+    addon = sub.promoted_addon.addon
+
+    # Create the Stripe URL pointing to the Stripe subscription.
+    stripe_sub_url = "https://dashboard.stripe.com"
```
Comment (diox) [inline src/olympia/promoted/tasks.py:85]: Nit: we don't enforce it so it's fine to keep as-is, but addons-server is using single quotes wherever possible

File: src/olympia/promoted/tasks.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    if not stripe_sub["livemode"]:
+        stripe_sub_url = stripe_sub_url + "/test"
+    stripe_sub_url = stripe_sub_url + f"/subscriptions/{subscription_id}"
```
Comment (diox) [inline src/olympia/promoted/tasks.py:88]: Nit:

```suggestion
    stripe_sub_url = f"{stripe_sub_url}/subscriptions/{subscription_id}"
```

Should work (and above as well)

File: src/olympia/lib/settings_base.py [hunk 3]
```diff
@@ -1128,6 +1128,7 @@ def get_db_config(environ_var, atomic_requests=True):
     'olympia.devhub.tasks.validate_file': {'queue': 'devhub'},
     'olympia.devhub.tasks.validate_upload': {'queue': 'devhub'},
     'olympia.files.tasks.repack_fileupload': {'queue': 'devhub'},
+    'olympia.promoted.tasks.on_stripe_charge_failed': {'queue': 'devhub'},
```
Comment (diox) [inline src/olympia/lib/settings_base.py:1131]: I wonder if that's the best queue for it... I suppose it's fine for now - ultimately I hope admins will not rely on this notification only but also periodically visit the Stripe dashboard...

**Review by diox — APPROVED** [review]
r+, just nits.

File: src/olympia/lib/settings_base.py (hunk 3 above)
Comment (willdurand) [reply src/olympia/lib/settings_base.py:1131]: Yeah, I don't know either.. I didn't want to create a new queue for that right now, though.

**Pull request merged**